### DISCUSSION TITLE
zfs sync=always, some questions about read operations

### DISCUSSION THREAD

**Discussion by unkownerror1** [openzfs/zfs#14994] [Q&A]
zfs sync=always。The zil_commit function is called whenever there are write and read operations. Why call the zil_commit function once when reading

**Comment by kftsehk** [comment]
Is it about `atime=on` thing that a read translates to an update in access time attribute?

**Reply by unkownerror1** [reply]
Thanks for the answer, but the read operation only reads user data, does it matter if atime is not updated？

**Reply by kftsehk** [reply]
if you have some requirements on accurate record of the last access time to the timestamp it is opened for read, even if it is not modified, for every file on the file system, then you must use `atime=on` and `relatime=off`

if you want relaxed record accuracy to within 24hr, maybe for housekeeping identifying files not used for years, turn `relatime=on` will only update if that file is accessed but not modified and last access is more than 24h ago.

or you can turn atime off if it doesn’t matter to you at all.

I might keep the relatime in a multi-user env so if I run out of disk space I can point finger to someone to cleanup their old project file on expensive storage.

it depends on your use case at hand
